src/json.c

```diff
@@ -287,15 +281,15 @@ lyjson_string_(struct lyjson_ctx *jsonctx)
     }
 
     /* EOF reached before endchar */
-    LOGVAL(jsonctx->ctx, LY_VLOG_LINE, &jsonctx->line, LY_VCODE_EOF);
-    LOGVAL(jsonctx->ctx, LY_VLOG_LINE, &start_line, LYVE_SYNTAX, "Missing quotation-mark at the end of a JSON string.");
+    LOGVAL(jsonctx->ctx, LY_VCODE_EOF);
+    LOGVAL_LINE(jsonctx->ctx, start_line, LYVE_SYNTAX, "Missing quotation-mark at the end of a JSON string.");
```

Comment:
To be honest, I've just kept the logic that was there. But it makes sense to me - in this particular place, the beginning of the string is being remembered to better refer where is probably the problem, because when you forget to close a string, it doesn't help much to report the last line of the input, the line where the string begins is much better.
Actually, I would expect to do the same at more places, but this is the only place we do it. So I consider using this macro as a possible future enhancement to provide more accurate/useful information.

Reply:
So the lines are counted normally except in this case even the starting line is reported? If so, then I have no problem with it, I misunderstood it before.

Reply:
yes, there are 2 messages, the first report unexpected end of data on the current/last line and the following message says that the input misses double-quote for the string started on some possibly earlier line. And since logging gets line number from the current position of the input handler, `LOGVAL_LINE` macro forces to use some specific line number instead. Newlines counting in the input handler is not affected.